### PR TITLE
HIVE-25093: date_format() UDF is returning values in UTC time zone only

### PR DESCRIPTION
**Pull request by ashish-kumar-sharma**

### CODE REVIEW

**Review by guptanikhil007 — COMMENTED** [review]
@ashish-kumar-sharma If other date udfs are using the UTC calendar then there might be difference between results of various date functions which will be unexpected from user standpoint.
Can you verify this for other functions?

**Comment by ashish-kumar-sharma** [comment]
> @ashish-kumar-sharma If other date udfs are using the UTC calendar then there might be difference between results of various date functions which will be unexpected from user standpoint.
> Can you verify this for other functions?

@guptanikhil007 UDF is performing as expected only thing is at the end user default session time zone is overridden with UTC

File: ql/src/java/org/apache/hadoop/hive/ql/util/DateTimeMath.java [hunk 1]
```diff
@@ -613,4 +615,18 @@ public static Calendar getProlepticGregorianCalendarUTC() {
...
+
+  /**
+   * TODO - this is a temporary fix for handling Julian calendar dates.
+   * Returns a Gregorian calendar that can be used from year 0+ instead of default 1582.10.15.
+   * This is desirable for some UDFs that work on dates which normally would use Julian calendar.
+   * @return the calendar
+   */
+  public static Calendar getTimeZonedProlepticGregorianCalendar() {
+    GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone(
+        SessionState.get() == null ? new HiveConf().getLocalTimeZone() : SessionState.get().getConf()
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/util/DateTimeMath.java:627]: Use () for condition.

Comment (ashish-kumar-sharma) [reply ql/src/java/org/apache/hadoop/hive/ql/util/DateTimeMath.java]: Done

**Review by adesh-rao — APPROVED** [review]

**Comment by zabetak** [comment]
I was really surprised to see that this PR which is around dates had changes in `HiveConnection` class. It appears that those were reverted in https://github.com/apache/hive/pull/2404 and thankfully there are comments describing the reasoning there. I am adding a comment here in case someone else in the future bumps into the same lines of code.

**Pull request merged**